### PR TITLE
fix: Fixed configuration of individual metric & log category diagnostic settings

### LINKED ISSUE #898: [AVM Module Issue]: Unable to enable `diagnosticSettings` by category - missing `enabled` boolean on diagnosticSettingType
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

Other, as defined below...

### (Optional) Module Name if not listed above

Every module

### (Optional) Module Version

_No response_

### Description

It is not possible to use a category name and enable it via the **diagnosticSettingType** for **AVM all modules**.

It appears the `logCategoriesAndGroups[].enabled` defaults to false. I assume the same is true for `metricCategories[].enabled` There is not a property for setting `enabled` unless you take the defaults of AllMetrics/AllLogs.

```bicep
    diagnosticSettings: [
      {
        name: 'toLA'
        workspaceResourceId: '.../providers/microsoft.operationalinsights/workspaces/...'
        logCategoriesAndGroups: [
          {
            category: 'PostgreSQLFlexSessions'
          }
          {
            category: 'PostgreSQLFlexQueryStoreRuntime'
          }
          {
            category: 'PostgreSQLFlexQueryStoreWaitStats'
          }
        ]
      }
    ]
```

Results in the following:
image

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Worked `enabled` flag into diagnostic settings deployment
- Regenerated all docs & templates (as I need anyways for this PR for all resources with diagnostic settings)

# Test cases
I tested pretty much any scenario, but here are a few examples

## Specific Log set + allMetrics, but disabled
```bicep
module kvlt '(...)/avm/res/key-vault/vault/main.bicep' = {
  name: 'alsehrdiagkv-deployment'
  params: {
    (...)
    diagnosticSettings: [
      {
        name: 'customSetting'
        logCategoriesAndGroups: [
          {
            category: 'AzurePolicyEvaluationDetails'
            enabled: true
          }
        ]
        metricCategories: [
          {
            category: 'AllMetrics'
            enabled: false
          }
        ]
        (...)
    ]
  }
}
```
image
 
## No specific log set & no specific metric set
```bicep
module kvlt '(...)/avm/res/key-vault/vault/main.bicep' = {
  name: 'alsehrdiagkv-deployment'
  params: {
    (...)
    diagnosticSettings: [
      {
        name: 'customSetting'
        (...)
    ]
  }
}
```
image

## 2 Logs set, one enabled, one disabled
```bicep
module kvlt '(...)/avm/res/key-vault/vault/main.bicep' = {
  name: 'alsehrdiagkv-deployment'
  params: {
    (...)
    diagnosticSettings: [
      {
        name: 'customSetting'
        logCategoriesAndGroups: [
          {
            category: 'AzurePolicyEvaluationDetails'
          }
          {
            category: 'AuditEvent'
            enabled: false
          }
        ]
  }
}
```
image

| Pipeline |
| - |
[avm.res.analysis-services.server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.analysis-services.server.yml)
[avm.res.api-management.service](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml)
[avm.res.app.managed-environment](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.app.managed-environment.yml)
[avm.res.automation.automation-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.automation.automation-account.yml)
[avm.res.batch.batch-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.batch.batch-account.yml)
[avm.res.cache.redis](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)
[avm.res.cognitive-services.account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.cognitive-services.account.yml)
[avm.res.compute.availability-set](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.availability-set.yml)
[avm.res.compute.disk-encryption-set](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.disk-encryption-set.yml)
[avm.res.compute.disk](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.disk.yml)
[avm.res.compute.gallery](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.gallery.yml)
[avm.res.compute.image](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.image.yml)
[avm.res.compute.proximity-placement-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.proximity-placement-group.yml)
[avm.res.compute.ssh-public-key](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.ssh-public-key.yml)
[avm.res.compute.virtual-machine](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine.yml)
[avm.res.consumption.budget](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.consumption.budget.yml)
[avm.res.container-registry.registry](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.container-registry.registry.yml)
[avm.res.container-service.managed-cluster](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)
[avm.res.data-factory.factory](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.data-factory.factory.yml)
[avm.res.data-protection.backup-vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.data-protection.backup-vault.yml)
[avm.res.databricks.access-connector](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.databricks.access-connector.yml) (already being worked on by @clintgrove)
[avm.res.databricks.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.databricks.workspace.yml) (cannot test in fork as different tenant)
[avm.res.db-for-my-sql.flexible-server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.db-for-my-sql.flexible-server.yml)
[avm.res.db-for-postgre-sql.flexible-server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.db-for-postgre-sql.flexible-server.yml)
[avm.res.desktop-virtualization.application-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.application-group.yml)
[avm.res.desktop-virtualization.host-pool](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.host-pool.yml)
[avm.res.desktop-virtualization.scaling-plan](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.scaling-plan.yml) (unrelated)
[avm.res.desktop-virtualization.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.desktop-virtualization.workspace.yml)
[avm.res.document-db.database-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml)
[avm.res.event-grid.domain](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.event-grid.domain.yml)
[avm.res.event-grid.system-topic](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.event-grid.system-topic.yml)
[avm.res.event-grid.topic](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.event-grid.topic.yml)
[avm.res.health-bot.health-bot](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.health-bot.health-bot.yml)
[avm.res.insights.action-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.action-group.yml)
[avm.res.insights.activity-log-alert](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.activity-log-alert.yml)
[avm.res.insights.component](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)
[avm.res.insights.data-collection-endpoint](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.data-collection-endpoint.yml)
[avm.res.insights.data-collection-rule](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.data-collection-rule.yml)
[avm.res.insights.diagnostic-setting](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.diagnostic-setting.yml)
[avm.res.insights.metric-alert](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.metric-alert.yml)
[avm.res.insights.private-link-scope](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.private-link-scope.yml)
[avm.res.insights.scheduled-query-rule](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.scheduled-query-rule.yml)
[avm.res.insights.webtest](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.webtest.yml)
[avm.res.key-vault.vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)
[avm.res.kubernetes-configuration.extension](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.extension.yml)
[avm.res.kubernetes-configuration.flux-configuration](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.flux-configuration.yml)
[avm.res.logic.workflow](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.logic.workflow.yml)
[avm.res.maintenance.maintenance-configuration](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.maintenance.maintenance-configuration.yml)
[avm.res.managed-identity.user-assigned-identity](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.managed-identity.user-assigned-identity.yml)
[avm.res.management.management-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.management.management-group.yml)
[avm.res.net-app.net-app-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)
[avm.res.network.application-security-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.application-security-group.yml)
[avm.res.network.bastion-host](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.bastion-host.yml)
[avm.res.network.connection](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.connection.yml)
[avm.res.network.ddos-protection-plan](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.ddos-protection-plan.yml)
[avm.res.network.dns-forwarding-ruleset](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.dns-forwarding-ruleset.yml)
[avm.res.network.dns-resolver](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.dns-resolver.yml)
[avm.res.network.dns-zone](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.dns-zone.yml)
[avm.res.network.express-route-circuit](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml)
[avm.res.network.express-route-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.express-route-gateway.yml)
[avm.res.network.firewall-policy](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.firewall-policy.yml)
[avm.res.network.front-door-web-application-firewall-policy](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.front-door-web-application-firewall-policy.yml)
[avm.res.network.front-door](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.front-door.yml)
[avm.res.network.ip-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.ip-group.yml)
[avm.res.network.load-balancer](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.load-balancer.yml)
[avm.res.network.local-network-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.local-network-gateway.yml)
[avm.res.network.nat-gateways](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.nat-gateway.yml)
[avm.res.network.network-interface](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.network-interface.yml)
[avm.res.network.network-security-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.network-security-group.yml)
[avm.res.network.private-dns-zone](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.private-dns-zone.yml)
[avm.res.network.private-endpoint](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.private-endpoint.yml)
[avm.res.network.private-link-service](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.private-link-service.yml)
[avm.res.network.public-ip-address](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.public-ip-address.yml)
[avm.res.network.route-table](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.route-table.yml)
[avm.res.network.trafficmanagerprofile](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.trafficmanagerprofile.yml)
[avm.res.network.virtual-hub](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.virtual-hub.yml)
[avm.res.network.virtual-network-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network-gateway.yml)
[avm.res.network.virtual-network](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network.yml)
[avm.res.network.virtual-wan](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.virtual-wan.yml)
[avm.res.network.vpn-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.vpn-gateway.yml)
[avm.res.network.vpn-site](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.vpn-site.yml)
[avm.res.operational-insights.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.operational-insights.workspace.yml)
[avm.res.operations-management.solution](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.operations-management.solution.yml)
[avm.res.power-bi-dedicated.capacity](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.power-bi-dedicated.capacity.yml)
[avm.res.resource-graph.query](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.resource-graph.query.yml)
[avm.res.resources.deployment-script](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.resources.deployment-script.yml)
[avm.res.resources.resource-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.resources.resource-group.yml)
[avm.res.search.search-service](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.search.search-service.yml)
[avm.res.service-bus.namespace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.service-bus.namespace.yml)
[avm.res.sql.server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.sql.server.yml)
[avm.res.storage.storage-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)
[avm.res.synapse.private-link-hub](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.synapse.private-link-hub.yml)
[avm.res.virtual-machine-images.image-template](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.virtual-machine-images.image-template.yml)
[avm.res.web.serverfarm](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.web.serverfarm.yml)
[avm.res.web.site](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.web.site.yml)
[avm.res.web.static-site](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.web.static-site.yml)

### CODE REVIEW

**Review by AlexanderSehr — COMMENTED** [review]
Note, version 0.23.53 seems latest

**Comment by eriqua** [comment]
@AlexanderSehr not for this PR, but should we plan for testing not default behavior in, e.g.,  max tests?

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr not for this PR, but should we plan for testing not default behavior in, e.g., max tests?

Some modules do, but not consistently. No reason to not change that :) @eriqua

**Review by ghost — DISMISSED** [review]
Looking fine for me. Off topic suggestion. Wouldn't be better if we add and use a diagnostic settings module? You had to make your change everywhere lol

**Review by krbar — DISMISSED** [review]

**Review by clintgrove — DISMISSED** [review]
Looks good

File: avm/res/network/nat-gateway/main.bicep [hunk 1]
```diff
@@ -203,41 +203,3 @@ type roleAssignmentType = {
   @description('Optional. The Resource Id of the delegated managed identity resource.')
   delegatedManagedIdentityResourceId: string?
 }[]?
-
```
Comment (eriqua) [inline avm/res/network/nat-gateway/main.bicep:206]: what happened here? also, is it expected not to have any README.md update?

File: avm/res/network/front-door/tests/e2e/max/main.test.bicep [hunk 2]
```diff
@@ -80,7 +80,6 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
               privateLinkAlias: ''
               privateLinkApprovalMessage: ''
               privateLinkLocation: ''
-              privateLinkResourceId: ''
```
Comment (eriqua) [inline avm/res/network/front-door/tests/e2e/max/main.test.bicep:83]: intended?

File: avm/res/network/front-door/tests/e2e/max/main.test.bicep [hunk 3]
```diff
@@ -193,12 +192,6 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
         storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
         workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
       }
-      {
```
Comment (eriqua) [inline avm/res/network/front-door/tests/e2e/max/main.test.bicep:196]: intended?

File: avm/res/network/virtual-network-gateway/tests/e2e/waf-aligned/main.test.bicep [hunk 4]
```diff
@@ -126,7 +126,7 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
       }
       {
         name: 'nat-rule-2-dynamic-EgressSnat'
-        type: 'Dynamic'
+        type: 'Static'
```
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/tests/e2e/waf-aligned/main.test.bicep:129]: intended?

File: avm/res/network/virtual-network-gateway/tests/e2e/max/main.test.bicep [hunk 5]
```diff
@@ -145,7 +145,7 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
       }
       {
         name: 'nat-rule-2-dynamic-EgressSnat'
-        type: 'Dynamic'
+        type: 'Static'
```
Comment (eriqua) [inline avm/res/network/virtual-network-gateway/tests/e2e/max/main.test.bicep:148]: intended?

File: avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep [hunk 6]
```diff
@@ -94,7 +94,7 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Key-Vault-Crypto-User-RoleAssignment')
   scope: keyVault::key
   properties: {
-    principalId: 'b9199ba0-d7a7-499e-a9c1-ee87112c5aec' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
```
Comment (eriqua) [inline avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep:97]: too bad

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep [hunk 7]
```diff
@@ -94,7 +94,7 @@ resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   name: guid('msi-${keyVault::key.id}-${location}-${managedIdentity.id}-Key-Key-Vault-Crypto-User-RoleAssignment')
   scope: keyVault::key
   properties: {
-    principalId: 'b9199ba0-d7a7-499e-a9c1-ee87112c5aec' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
+    principalId: '711330f9-cfad-4b10-a462-d82faa92027d' // AzureDatabricks Enterprise Application Object Id (Note: this is tenant specific)
```
Comment (eriqua) [inline avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep:97]: oh no

File: avm/res/key-vault/vault/tests/e2e/max/main.test.bicep [hunk 8]
```diff
@@ -116,12 +116,6 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
         storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
         workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
       }
-      {
```
Comment (eriqua) [inline avm/res/key-vault/vault/tests/e2e/max/main.test.bicep:119]: intended?

File: avm/res/compute/virtual-machine/main.bicep [hunk 9]
```diff
@@ -740,19 +740,25 @@ type diagnosticSettingType = {
   @description('Optional. The name of diagnostic setting.')
   name: string?
 
-  @description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource. Set to \'\' to disable log collection.')
+  @description('Optional. The name of logs that will be streamed. "allLogs" includes all possible logs for the resource. Set to `[]` to disable log collection.')
```
Comment (eriqua) [inline avm/res/compute/virtual-machine/main.bicep:743]: Shouldn't this reflect in a README.md change? I don't see it for vm

File: avm/res/cognitive-services/account/tests/e2e/max/main.test.bicep [hunk 10]
```diff
@@ -88,12 +88,6 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
         storageAccountResourceId: diagnosticDependencies.outputs.storageAccountResourceId
         workspaceResourceId: diagnosticDependencies.outputs.logAnalyticsWorkspaceResourceId
       }
-      {
```
Comment (eriqua) [inline avm/res/cognitive-services/account/tests/e2e/max/main.test.bicep:91]: intended?

File: avm/res/network/front-door/tests/e2e/max/main.test.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply avm/res/network/front-door/tests/e2e/max/main.test.bicep:83]: Yes, fixes test

File: avm/res/network/front-door/tests/e2e/max/main.test.bicep (hunk 3 above)
Comment (AlexanderSehr) [reply avm/res/network/front-door/tests/e2e/max/main.test.bicep:196]: Yes, fixes test (see comment to your orginal PE test comment)

File: avm/res/network/front-door/tests/e2e/waf-aligned/main.test.bicep [hunk 11]
```diff
@@ -71,7 +71,6 @@ module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem'
               privateLinkAlias: ''
               privateLinkApprovalMessage: ''
               privateLinkLocation: ''
-              privateLinkResourceId: ''
```
[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/res/network/front-door/tests/e2e/waf-aligned/main.test.bicep:74]: Yes fixes test

File: avm/res/network/virtual-network-gateway/tests/e2e/max/main.test.bicep (hunk 5 above)
Comment (AlexanderSehr) [reply avm/res/network/virtual-network-gateway/tests/e2e/max/main.test.bicep:148]: Yes, fixes test

File: avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep (hunk 6 above)
Comment (AlexanderSehr) [reply avm/res/databricks/workspace/tests/e2e/max/dependencies.bicep:97]: Too bad indeed. This is the AVM one (hence the pipeline is red).
Mind you - I did try to fetch the ID via an MSI+DS but as expected it doesn't have the permissions to do that.

File: avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep (hunk 7 above)
Comment (AlexanderSehr) [reply avm/res/databricks/workspace/tests/e2e/waf-aligned/dependencies.bicep:97]: See other comment. Nonono indeed

File: avm/res/key-vault/vault/tests/e2e/max/main.test.bicep (hunk 8 above)
Comment (AlexanderSehr) [reply avm/res/key-vault/vault/tests/e2e/max/main.test.bicep:119]: Yes, fixes test (see comment to your orginal PE test comment)

File: avm/res/compute/virtual-machine/main.bicep (hunk 9 above)
Comment (AlexanderSehr) [reply avm/res/compute/virtual-machine/main.bicep:743]: Will doublecheck

File: avm/res/cognitive-services/account/tests/e2e/max/main.test.bicep (hunk 10 above)
Comment (AlexanderSehr) [reply avm/res/cognitive-services/account/tests/e2e/max/main.test.bicep:91]: Yes, fixes test (see comment to your orginal PE test comment)

File: avm/res/compute/virtual-machine/main.bicep (hunk 9 above)
Comment (AlexanderSehr) [reply avm/res/compute/virtual-machine/main.bicep:743]: ok checked. The VM itself does not support diagnostic settings but the contained NIC deployment does. Will remove it.

Comment (AlexanderSehr) [reply avm/res/compute/virtual-machine/main.bicep]: Fixed / updated only in inner NIC template

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**